xde/xde-link/build.rs

```diff
@@ -0,0 +1,53 @@
+fn main() {
+    // Match linker flags used with other devfsadm plugins.
+    //
+    // DYNFLAGS = $(HSONAME) $(ZTEXT) $(ZDEFS) $(BDIRECT) \
+    //      $(MAPFILES:%=-Wl,-M%) $(MAPFILE.PGA:%=-Wl,-M%) $(MAPFILE.NED:%=-Wl,-M%) \
+    //      $(LDCHECKS)
+    //
+    // See:
+    // - usr/src/Makefile.master
+    // - usr/src/lib/Makefile.lib
+    // - usr/src/cmd/devfsadm/Makefile.com
+
+    // $(HSONAME)
+    println!("cargo:rustc-cdylib-link-arg=-Wl,-hSUNW_xde_link.so");
+
+    // $(ZTEXT) $(ZDEFS) $(BDIRECT)
+    println!("cargo:rustc-cdylib-link-arg=-Wl,-ztext");
+    println!("cargo:rustc-cdylib-link-arg=-Wl,-zdefs");
+    println!("cargo:rustc-cdylib-link-arg=-Wl,-Bdirect");
+
+    // $(MAPFILES)
+    //
+    // We reference symbols that exist only within devfsadm itself and
+    // can only be resolved at runtime. At link time though it remains
+    // unresolved and -zdefs thus forces an error. We suppress the error
+    // by explicitly telling the linker that these symbols are external
+    // via a mapfile (map.devfsadm-externs).
+    //
+    // See usr/src/cmd/devfsadm/mapfile-vers
+    println!("cargo:rustc-cdylib-link-arg=-Wl,-Mmap.devfsadm-externs");
```

Comment:
Longer term this needs to be built against the actual mapfile and probably we shouldn't define our own given that this is private. I'm not suggesting we change it now, but something for us to be aware of. cc @jclulow